### DISCUSSION TITLE
can i change datetime picker into month picker

### DISCUSSION THREAD

**Discussion by ayemin-tun** [filamentphp/filament#9302] [General]
### Description 

can i change default datepicker into only show month and year
![Screenshot (135)](https://github.com/filamentphp/filament/assets/76741291/70bb24a8-5ee7-4fa0-8ab5-f6b3c2b292b5)
this datepicker into This 
![Screenshot (136)](https://github.com/filamentphp/filament/assets/76741291/f08e6a4a-336c-4858-8117-f40f172a0790)

**Comment by ArtMin96** [comment]
```html
<input type="month" />
```

```php
TextInput::make('field')
    ->type('month')
```

**Reply by tominal** [reply]
https://github.com/filamentphp/filament/discussions/9302#discussioncomment-9318183

**Comment by underdpt** [comment]
For anybody getting here, please check browser support. As of today, Firefox and Safari doesn't support it:

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/month#browser_compatibility

**Comment by Balachandran12** [comment]
I'm trying to fill the value on this field. But I can't able to fill or set default value.

**Reply by franmesa** [reply]
->default(now())

**Reply by ucosatti** [reply]
I have same issue, but default(now()) doesnt work.

**Reply by franmesa** [reply]
show code

**Reply by ucosatti** [reply]
```
                            DatePicker::make("period")
                                ->required()
                                ->native()
                                ->default(now())
                                ->extraInputAttributes([
                                    'type' => 'month',
                                    "min" => Carbon::now()->subYears(10)->format("Y-m"),
                                    "max" => Carbon::now()->subMonth()->format("Y-m"),
                                ]),
```
<img width="534" alt="image" src="https://github.com/user-attachments/assets/fa649875-350e-4e27-9601-e036a3baa5c0" />

**Reply by franmesa** [reply]
use Textinput instead of datetimepicker

hope this solve

**Reply by ucosatti** [reply]
nop, still the same. The issue is with type= month, even passing _value_ extra atribute

**Reply by franmesa** [reply]
try this out..
 TextInput::make('month')
                ->type('month')
                ->hiddenLabel()
                ->default(now()->format('Y-m')),

**Reply by ucosatti** [reply]
👏👏👏 You got it! Thank you very much!

**Comment by juanjoseva** [comment]
DatePicker::make('startDate')
    ->native()
    ->extraInputAttributes(['type' => 'month'])

**Reply by tominal** [reply]
https://github.com/filamentphp/filament/discussions/9302#discussioncomment-9318183

**Reply by juanjoseva** [reply]
my problem is when I retrieve the date. When I'm in edition mode the date does not show at all.

**Reply by Danielopes7** [reply]
> my problem is when I retrieve the date. When I'm in edition mode the date does not show at all.

same here

**Comment by mokhosh** [comment]
@juanjoseva @Danielopes7 

```php
->formatStateUsing(fn ($state) => Carbon::parse($state)->format('Y-m'))
```

**Reply by PhilippeDirx** [reply]
You should check if the $state is filled before parsing it with Carbon, unless you always want it to show the current month by default
